csfieldguide/chapters/content/en/coding-compression/sections/general-purpose.md

```diff
@@ -43,27 +31,337 @@ Here are some of the bits from the example earlier in this chapter; you can past
 111110000011111
 ```
 
-In fact, this is essentially what happens with GIF and PNG images; the pixel values are compressed using the Ziv-Lempel algorithm, which works well if you have lots of consecutive pixels the same colour.
-But it works very poorly with photographs, where pixel patterns are very unlikely to be repeated.
+What you just tried is essentially what happens with PNG images; the pixel values are compressed using a similar Ziv-Lempel algorithm, which works well if you have lots of consecutive pixels that are the same colour.
+But it works very poorly with photographs, where pixel patterns are very unlikely to be repeated exactly.
+For example, in the following row of pixels from a photo nearly all of them are the same colour, yet the number representing the colour value fluctuates slightly, so there aren't many repeated patterns that an LZ method would pick up:
+
+{image file-path="img/chapters/pixel-viewer-photo-values.png" alt="Row from pixel viewer interactive showing different colours in a photo." caption="true"}
+
+This image from a photo was generated using the Pixel Viewer Interactive.
+
+{image end}
+
+In contrast, the following zoomed in image from a cartoon has a lot of repeated pixel colours:
+
+{image file-path="img/chapters/pixel-viewer-cartoon-values.png" alt="Row from pixel viewer interactive showing similar colours in a cartoon." caption="true"}
+
+This image from a cartoon was generated using the Pixel Viewer Interactive.
+
+{image end}
+
+{panel type="teacher-note"}
+
+# Teacher note
+
+Ask your students to use the Pixel Viewer Interactive to load a photo and a cartoon to see this for themselves.
 
-{comment xtcb extra for experts: compress "aaaaaaaaaa". how can it decode? .. xtcb  Project .. Students should try an algorithm [zip, gzip etc.] on several different text files, image files, sound files, and video files. .. What would happen if text compression was lossy? (I think this should be a part of their merit discussion)}
+{panel end}
+
+In the next section we’ll take a look at a variation of “Ziv-Lempel coding” creatively called “Lempel-Ziv-Welch coding”, which is often referred  to as LZW to make things easier.
+LZW uses a dictionary-like structure based on key and value pairs - these are based on the metaphor of looking up a word (key) and its definition (value) in a dictionary, but in this case the key is a number, and the value is a phrase of text that it represents.
```

Comment:
`pairs - these` needs an ndash. Look for this in other places (I ctrl-f `space-hyphen-space` to find them)